From 87b7420ce58e6770c884aeaf5893470ff461f07e Mon Sep 17 00:00:00 2001
From: Lukasz Bartosik <lb@semihalf.com>
Date: Wed, 17 Oct 2018 14:51:02 +0300
Subject: [PATCH 0290/1051] octeontx: fix VF selection for TIM

In case of multiple domains when application was using TIM coprocessor
and running on subsequent domains then VF for TIM was not correctly
selected.

Signed-off-by: Lukasz Bartosik <lb@semihalf.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/timpf_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/timpf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/timpf_main.c
index c1adbf4d9606..d63bc1171d45 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/timpf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/timpf_main.c
@@ -169,7 +169,7 @@ static struct timpf *tim_dev_from_ringid(int id, int domain_id,
 			if (vf->domain.domain_id == domain_id &&
 			    vf->domain.subdomain_id == ringid) {
 				spin_unlock(&octeontx_tim_dev_lock);
-				*ring = ring_from_ringid(ringid);
+				*ring = i;
 				return tim;
 			}
 		}
-- 
2.17.1

